realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -145,8 +145,12 @@ public RealmObjectSchema addField(String fieldName, Class<?> fieldType, FieldAtt
                 containsAttribute(attributes, FieldAttribute.PRIMARY_KEY)) {
             nullable = false;
         }
-        long columnIndex = table.addColumn(metadata.realmType, fieldName, nullable);
+        // allow String & boxed variant type PrimaryKey to be null
+        if (containsAttribute(attributes, FieldAttribute.PRIMARY_KEY) && isNullablePrimaryKeyType(metadata)) {
```

Comment:
This seems a bit strange. Shouldn't it be enough to just remove `containsAttribute(attributes, FieldAttribute.PRIMARY_KEY)` from the above?